addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -5200,6 +5207,34 @@ registry.layout_column = SnippetOptionWidget.extend({
         gridUtils._resizeGrid(rowEl);
         this.trigger_up('activate_snippet', {$snippet: $(newColumnEl)});
     },
+    /**
+     * @override
+     */
+    async selectStyle(previewMode, widgetValue, params) {
+        const rowEl = this.$target[0];
+        if (params.cssProperty === "column-gap") {
+            // Limit the column-gap so the row does not overflow: the gap cannot
+            // be bigger than the grid column size.
+            widgetValue = `${Math.min(parseFloat(widgetValue), rowEl.clientWidth / 23)}px`;
```

Comment:
> You could round the value

Yes, I was not sure it would be justified so I let the real value and waited for an opinion about it :slightly_smiling_face: 

> but really I think we can just hardcode a maximum value, like 60px and it will be ok, no?

I thought of using a hardcoded value but the issue is I didn't know which value to use because the columns size is relative to the screen width (unlike the rows that are hardcoded at 50px) So there is always a point where the column gap could be too big :confused: 
Here at worst, when the screen will be small enough such that the columns size will be 0, there is a "guarantee" that the total width would still not overflow (this is what I had in mind when setting the ratio but I agree it is overkill :sweat_smile:) 
The other reason I used the column size as a limit is that if a user needs to put a gap bigger than the column size, then he could just move/resize the grid item by one row instead of changing the gap :shrug: 

But now that I think of it, I could use the row width when it is "container small" and divide it by 11 (gaps) (it seems to be a good upper bound) so it gives 690 / 11 = 62.73 px => rounded down it is 62px
But then 60px is a prettier number haha so yes in the end I agree with you => I will put 60px :slightly_smiling_face:  

> make that a InputUserValueWidget parameter

I wanted to add a `data-max` attribute to the input indeed, but as it was not a hardcoded value, I didn't know how to make it "dynamic", which is why I did it in `selectStyle` :slightly_smiling_face:

Reply:
Ah... we can't add `data-max` to inputs ? :open_mouth: It seems to only apply to ranges :grimacing: Should I add the support for that ?

Reply:
Yep that's what I meant by "maybe a good idea would be to make that a InputUserValueWidget parameter", I saw this was not yet a thing, would be a good thing to add :+1: